tests/lib/prepare-uc20.sh

```diff
@@ -0,0 +1,281 @@
+#!/bin/bash
+
+set -e
+set -x 
+
+apt update -yqq
+
+# these should already be installed in GCE images with the google-nested 
+# backend, but in qemu local images from qemu-nested, we might not have them
+apt install snapd ovmf qemu-system-x86 sshpass whois -yqq
+
+# use the snapd snap explicitly
+# TODO: since ubuntu-image ships it's own version of `snap prepare-image`, 
+# should we instead install beta/edge snapd here and point ubuntu-image to this
+# version of snapd?
+snap install snapd
+
+# install some dependencies
+# TODO:UC20: when should we start using candidate / stable ubuntu-image?
+snap install ubuntu-image --edge --classic
+
+# needed for dracut which is a build-dep of ubuntu-core-initramfs
+add-apt-repository ppa:snappy-dev/image -y
+
+# install build-deps for ubuntu-core-initramfs
+(
+    cd "$SETUPDIR"
+    apt -y build-dep ./
+)
+
+# TODO: is this still necessary for core-initrd? (this was copied from snapd)
+# create test user for spread to use
+groupadd --gid 12345 test
+adduser --uid 12345 --gid 12345 --disabled-password --gecos '' test
+
+if getent group systemd-journal >/dev/null; then
+    usermod -G systemd-journal -a test
+    id test | MATCH systemd-journal
+fi
+
+echo 'test ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
+
+# get the model
+# TODO: should this be a variant to test multiple grades i.e. signed and secured?
```

Comment:
looking at this again, it occurs to me that we can't specify unasserted snaps for signed or secured models, which means we couldn't inject users into the VM's for those grades, so probably this TODO should just go away and we are stuck with dangerous only grades.

restricting ourselves to just dangerous also means we could just use cloud-init instead of the whole special user setup too :thinking: